csrc/libcrypto_rng.h

```diff
@@ -0,0 +1,35 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef ACCP_LIBCRYPTO_RNG_H
+#define ACCP_LIBCRYPTO_RNG_H 1
+
+#include <openssl/evp.h>
+#include <openssl/rand.h>
+#include <vector>
+#include "generated-headers.h"
+#include "util.h"
+#include "compiler.h"
+
+namespace AmazonCorrettoCryptoProvider {
+
+class libcrypto_rng {
```

Comment:
We clearly don't need to track state anymore, so rather than have the complexity of a class and then needing to track this state on the Java side (which can be expensive), let's remove *all* native state and just call directly into `RAND_bytes`.

Reply:
Fixed